File: python/2020/day23/p1.py
from common import ll
n = list(map(int, '496138527'))
(h, d) = ll.make(n)

c = ll.run(h, d, 100, 9)
while c.data != 1:
    c = c.next
s = ''
while c.next.data != 1:
    c = c.next
    s += str(c.data)
print(s)

File: haskell/2015/day2/p1.hs
import Data.Common

calc :: [Int] -> Int
calc [w, l, h] =
    let lw = l * w
        lh = l * h
        wh = w * h
    in (2 * lw) + (2 * lh) + (2 * wh) + (minimum [lw, lh, wh])

main = do
    contents <- getContents
    putStrLn $ solve calc contents
